src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -407,6 +408,28 @@ private void createMesosTask(Offer offer, Request request) {
     LOGGER.info("Launching task " + taskId.getValue() + " with URI " +
                 joinPaths(jenkinsMaster, SLAVE_JAR_URI_SUFFIX));
 
+    for (final Computer computer : Jenkins.getInstance().getComputers()) {
```

Comment:
It's possible to receive an offer to launch the same mesos task after it has already been launched. There's nothing preventing that from happening right now and it leads to orphans. We've seen tasks that have launched in mesos but never get killed and don't do anything because they launch after it has already been launched. These if statements guard from that race condition

On June 21, 2015 3:38:04 PM CDT, Vinod Kone <notifications@github.com> wrote:
>> @@ -407,6 +408,28 @@ private void createMesosTask(Offer offer,
>Request request) {
>>      LOGGER.info("Launching task " + taskId.getValue() + " with URI "
>+
>>                  joinPaths(jenkinsMaster, SLAVE_JAR_URI_SUFFIX));
>>  
>> +    for (final Computer computer :
>Jenkins.getInstance().getComputers()) {
>
>Sorry for not being clear. Yes, the id of the mesos task is the same as
>the name of the Jenkins slave and this was done to make it easy to
>figure out which mesos task to kill when the jenkins slave goes idle.
>
>What I am confused about is that, this code suggests we are creating
>duplicate task ids which is unexpected. Can you elaborate what do you
>mean by your second sentence?
>
>Thanks.
>
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/jenkinsci/mesos-plugin/pull/121/files#r32897375

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.